packages/yew/src/html/component/mod.rs

```diff
@@ -28,89 +48,37 @@ pub trait Component: Sized + 'static {
     /// commonly use an enum to declare multiple Message types.
     type Message: 'static;
 
-    /// Properties are the inputs to a Component and should not mutated within a
-    /// Component. They are passed to a Component using a JSX-style syntax.
-    /// ```
-    ///# use yew::{Html, Component, Properties, ComponentLink, html};
-    ///# struct Model;
-    ///# #[derive(Clone, Properties)]
-    ///# struct Props {
-    ///#     prop: String,
-    ///# }
-    ///# impl Component for Model {
-    ///#     type Message = ();type Properties = Props;
-    ///#     fn create(props: Self::Properties,link: ComponentLink<Self>) -> Self {unimplemented!()}
-    ///#     fn update(&mut self,msg: Self::Message) -> bool {unimplemented!()}
-    ///#     fn change(&mut self, _: Self::Properties) -> bool {unimplemented!()}
-    ///#     fn view(&self) -> Html {
-    /// html! {
-    ///     <Model prop="value" />
-    /// }
-    ///# }}
-    /// ```
+    /// The Component's properties.
+    ///
+    /// When the parent of a Component is re-rendered, it will either be re-created or
+    /// receive new properties in the context passed to the `changed` lifecycle method.
     type Properties: Properties;
 
-    /// Components are created with their properties as well as a `ComponentLink` which
-    /// can be used to send messages and create callbacks for triggering updates.
-    fn create(props: Self::Properties, link: ComponentLink<Self>) -> Self;
+    /// Called when component is created.
+    fn create(ctx: &Context<Self>) -> Self;
 
+    /// Called when a new message is sent to the component via it's scope.
+    ///
     /// Components handle messages in their `update` method and commonly use this method
     /// to update their state and (optionally) re-render themselves.
-    fn update(&mut self, msg: Self::Message) -> ShouldRender;
+    fn update(&mut self, _ctx: &Context<Self>, _msg: Self::Message) -> ShouldRender {
```

Comment:
Small nit: 
When you use IDE auto completion for trait methods in an `impl` block, it will copy this exact line with the leading underscores in the argument names. I'd remove the underscores and add `#[allow(unused_variables)]` instead, so you don't have to remove them each time after auto completion. 

Same for the other methods, of course.

Reply:
Also, as this is a good opportunity, do you (and other contributors) think that `ShouldRender` should be kept as an alias? I don't see much benefit in it over a comment on what the returned `bool` means. Furthermore, it has been rather annoying to either import another symbol or replace it with a bool after auto completion.